### ISSUE TITLE
Reduce typing friction when adding many Latex equations

### ISSUE THREAD

**Issue by withattribution** [Roam-Research/issues#388]
When adding several equations typing /Latex becomes tiresome. Since wrapping equation-text with a pair of $$ produces a latex formatted section I propose that typing two $$ automatically completes to ```$$ | $$``` with the cursor in the center mirroring the same functionality found when typing double brackets eg. [[ completes to ```[[ | ]]``` with the cursor in the middle ready to type.

A dedicated keyboard shortcut could be a simple way to go as well but in my opinion the above feels like idiomatic Roam.

**Comment by LuisMcLovin** [comment]
Yes this and also LaTeX code blocks! When I wish to do long-form maths or large formulae that need many lines there’s no support for that, yet there is for code blocks going several lines.

**Comment by panterarocks49** [comment]
Deployed a fix for this today
typing `$$|` results in `$$|$$`
highlighting text and typing $ will wrap it in double $$

@LuisMcLovin what do you mean by latex code blocks?
Do you mean code highlighting in our code block, like [this](https://codemirror.net/mode/stex/index.html)?

**Comment by michael-erasmus** [comment]
> Do you mean code highlighting in our code block, like this?

I've been using Roam for taking notes on Math lectures as well and this is something that I'd love to have as well!!

Also the ability to use packages: https://www.latex-tutorial.com/tutorials/packages/

There are many possible packages that can be installed, but there are many common ones that would be awesome to have installed by default:

This top 10 is a good list: http://blog.analogmachine.org/2011/05/16/top-10-latex-packages/

Also:
https://tex.stackexchange.com/questions/553/what-packages-do-people-load-by-default-in-latex

**Comment by LuisMcLovin** [comment]
> Deployed a fix for this today
> typing `$$|` results in `$$|$$`
> highlighting text and typing $ will wrap it in double $$
> 
> @LuisMcLovin what do you mean by latex code blocks?
> Do you mean code highlighting in our code block, like [this](https://codemirror.net/mode/stex/index.html)?

Hi pantera, Yes, indeed this is what was meant by a Latex code block :)

Lets me do math along several lines, just like how code blocks lets you do programming across several lines

If I want to do long-form math in Roam its a chore, as I'd have to ensure I $$ on every line. $$ Could simply be the start of a Latex *BLOCK*
